cli/command/node/list.go

```diff
@@ -45,21 +45,29 @@ func newListCommand(dockerCli *command.DockerCli) *cobra.Command {
 
 func runList(dockerCli *command.DockerCli, opts listOptions) error {
 	client := dockerCli.Client()
+	out := dockerCli.Out()
 	ctx := context.Background()
 
+	var info types.Info
+
 	nodes, err := client.NodeList(
 		ctx,
 		types.NodeListOptions{Filter: opts.filter.Value()})
 	if err != nil {
 		return err
 	}
 
-	info, err := client.Info(ctx)
+	if len(nodes) == 0 {
+		// if there is no node, no need to call /api endpoint and print directly.
+		goto SKIP
+	}
+
+	info, err = client.Info(ctx)
```

Comment:
Shouldn't `Info` also be skipped if `opts.quiet` is true? How about something like:

```go
if len(nodes) > 0 && !opts.quiet {
        info, err = client.Info(ctx)
        if err != nil {
                return err
        }
}
```